roles/custom/matrix-user-verification-service/tasks/main.yml

```diff
@@ -0,0 +1,24 @@
+---
+
+- name: verify all necessary variables are present
+  assert:
+    that:
+      - matrix_user_verification_service_uvs_access_token is defined and matrix_user_verification_service_uvs_access_token|length
+      - matrix_user_verification_service_uvs_homeserver_url is defined and matrix_user_verification_service_uvs_homeserver_url|length
+    fail_msg: "Missing variable in {{ matrix_user_verification_service_ansible_name }} role"
```

Comment:
It's better to move these validation checks into a `validate_config.yml` file here, like we do for other roles.

Also, the `validate_config.yml` tasks for the role should only run when the role is enabled (`matrix_user_verification_service_enabled`).

Dropped `is defined` checks (as discussed in other PR comments) would be better. All variables referenced by roles should have default values in `defaults/main.yml`. A variable being undefined should not be a possibility - it makes code like this ugly and discovery of variables hard. All variables should be defined and should have documentation around them showing what the default value is (and potentially why), what values they could take, what variables are related to them, links to upstream documentation (if applicable and useful), etc.

It's also a good time to think of any validation checks were forgotten to be added here.

I got the impression that this the matrix user verification service can only work with Synapse, not with other homeserver implementations? If so, it may make sense to add some check for `matrix_homeserver_implementation` here. If `matrix_homeserver_implementation != 'synapse'`, tell the user. This variable is part of `matrix-base` and we generally consider it OK to reach into `matrix-base` variables from any role. If you'd like to keep things more decoupled and pass it around through another variable (via `group_vars/matrix_servers`), that's fine too - it just makes things more complicated.